src/ui/public/vis/vis.js

```diff
@@ -40,6 +40,13 @@ import { queryManagerFactory } from '../query_manager';
 import { SearchSourceProvider } from '../courier/data_source/search_source';
 import { SavedObjectsClientProvider } from '../saved_objects';
 
+const getTerms = (table, columnIndex, rowIndex) => {
+  // get only rows where cell value matches current row for all the fields before columnIndex
+  const rows = table.rows.filter(row => row.every((cell, i) => cell === table.rows[rowIndex][i] || i >= columnIndex));
+  const terms = rows.map(row => row[columnIndex]);
+  return [...new Set(terms.filter(term => term !== '__other__'))];
```

Comment:
If you have either Other and Missing buckets, this function doesn't handle the case and add to the filter a `__missing__` field.

<img width="1255" alt="screen shot 2018-06-19 at 15 14 02" src="https://user-images.githubusercontent.com/1421091/41599433-7821c2c2-73d3-11e8-8324-28ee4681640a.png">

Reply:
And subsequent question: why if we filter by other, we are also filtering by "missing"?

Reply:
if Missing was one of the buckets (for example 400, 404, Missing) then Other means not 400, non 404 and not missing (so exists)

Reply:
Right, I was considering missing as not part of the "whole" since you can show/hide missing from the editor.